tests/gis_tests/gdal_tests/test_geom.py

```diff
@@ -320,6 +320,11 @@ def test_srs(self):
                     self.assertEqual('WGS 72', ring.srs.name)
                     self.assertEqual(4322, ring.srid)
 
+            # test assigning of SRS and SRID to their own values
+            mpoly = OGRGeometry(mp.wkt, srs=None)
+            mpoly.srs = mpoly.srs
+            mpoly.srid = mpoly.srid
```

Comment:
I find it more expressive. It's obvious that attribute should allow assigning to its own value, unlike that attribute should allow assigning to None.